### PR TITLE
feat(mobile): darken screen on backup page

### PR DESCRIPTION
**Pull request by Saschl**
Follow up of: https://github.com/immich-app/immich/pull/11621, will darken the screen when starting the backup to avoid burn-in issues.

I've decided to create 2 separate PRs as this one will likely take a bit longer as I need some feedback on how to implement this best UX wise.

Tested with Android emulator (Pixel 8), iOS Emulator (iPhone 15) and iPhone15 Plus, OnePlus 3T (real devices)

Current idea (sped up, actual timeout is 60 seconds), open for feedback :)

https://github.com/user-attachments/assets/42dfe652-cc20-447d-9c38-690df5e0000e

### CODE REVIEW

**Comment by alextran1502** [comment]
I think we don't need to show the confirmation form here. When the backup process start, we start a timer, after 1 minutes then we auto darken the screen

**Comment by Saschl** [comment]
> I think we don't need to show the confirmation form here. When the backup process start, we start a timer, after 1 minutes then we auto darken the screen

Sounds fine, applied the changes and removed the dialog.

File: mobile/lib/pages/backup/backup_controller.page.dart [hunk 1]
```diff
@@ -71,6 +79,30 @@ class BackupControllerPage extends HookConsumerWidget {
       [backupState.backupProgress],
     );
 
+    useEffect(
+      () {
+        if (backupState.backupProgress == BackUpProgressEnum.inProgress) {
+          darkenScreenTimer.value = Timer(const Duration(seconds: 60), () {
```
Comment (alextran1502) [inline mobile/lib/pages/backup/backup_controller.page.dart:85]: can we refactor this to a method so we don't duplicate code on line `306`?

**Comment by alextran1502** [comment]
Thank you for the contribution again! I tried it and like it a lot. One small request: Can we make the screen darken to a certain level, not completely dark? Think of an always-on mode screen shade so the user can still see what is happening?

**Review by alextran1502 — APPROVED** [review]

**Comment by Saschl** [comment]
Thanks for taking care! I initially chose to turn the screen black completely oled displays will essentially be off and avoid burn ins. But I’m not even sure if that’s a thing anymore.

**Comment by mortenmoulder** [comment]
Is there a toggle to disable this? If I want my screen to dim, I will do that myself. This kind of "feature" should definitely be opt-out by the very least.

**Comment by VibeEly** [comment]
Burn in on an OLED can set in after just 18 hours of static elements on screen, measured in YouTube endurance tests of flagship phones like the iPhone and Samsung Galaxy. If a user is doing a full phone backup, for example 100 GB, at Wi-Fi upload speeds, that can easily be a multi-hour task. The elements on the upload page are particularly static. 

A full black screen option for upload would prevent burn in on OLED screens during long phone backups from the mobile app. Examples of other mobile apps that black out the screen are time lapse photography apps. They often leave the phone on for long periods, such as overnight, which can burn out the screen, so this feature is useful for them. 

Currently, this can be mitigated by the user manually limiting uploads to brief periods. However a "tap to darken" button that completely blacks out all elements, then when you tap the screen again, they re-appear, would be a great feature that would fully solve the issue.

**Pull request merged**